e2e/tests-dfx/call.bash

```diff
@@ -298,3 +298,82 @@ teardown() {
   assert_command dfx canister call inter2_mo read
   assert_match '(8 : nat)'
 }
+
+@test "impersonate sender" {
+    [[ ! "$USE_POCKETIC" ]] && skip "skipped for replica: impersonating sender is only supported for PocketIC"
```

Comment:
Can we have this test impersonate an identity other than (or in addition to) aaaaa-aa? It looks like these tests would pass if `--impersonate` always used the anonymous identity rather than the impersonation mechanism.

Reply:
The principal `aaaaa-aa` is the _management canister_ identity, not the anonymous identity `2vxsx-fae`, and thus `--identity anonymous` would not make the tests pass.

Reply:
Okay, but that being the case, this test shows that `--impersonate` can impersonate the management canister identity

Reply:
> this test shows that --impersonate can impersonate the management canister identity

indeed and the impersonation mechanism is the only way how to make the management canister identity be the caller